sql: use crtime.Mono for phase times

RaduBerinde opened the pull request:
#### go.mod: update crlib dependency

Epic: none
Release note: None

#### sql/sessionphase: separate SessionInit time

The `SessionInit` is not really a phase, it is set just once.
This change separates it out from the phase times (which are used to
measure elapsed times).

Epic: none
Release note: None

#### sql: use crtime.Mono for phase times

Using just the monotonic timer is faster. When we need to convert to a
real `time.Time`, we use `now` as the reference time.

Informs: #133315
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/136911)
<!-- Reviewable:end -->

RaduBerinde committed:
sql: use crtime.Mono for phase times

Using just the monotonic timer is faster. When we need to convert to a
real `time.Time`, we use `now` as the reference time.

Informs: #133315
Release note: None

RaduBerinde committed:
go.mod: update crlib dependency

Epic: none
Release note: None

RaduBerinde committed:
sql/sessionphase: separate SessionInit time

The `SessionInit` is not really a phase, it is set just once.
This change separates it out from the phase times (which are used to
measure elapsed times).

Epic: none
Release note: None

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1, 3 of 3 files at r2, 17 of 17 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136911#-:-ODhNHNd10faDcyoi5sH:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dhartunian)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+
